File: archivist/src/main/java/com/zorroa/archivist/security/InternalAuthentication.java
package com.zorroa.archivist.security;

import com.zorroa.security.UserAuthed;
import org.springframework.security.authentication.AbstractAuthenticationToken;
import org.springframework.security.core.GrantedAuthority;

import java.util.Collection;

/**
 * InternalAuthentication is for internal server threads.
 */
public class InternalAuthentication extends AbstractAuthenticationToken {

    private final Object principal;

    public InternalAuthentication(UserAuthed user) {
        super(user.getAuthorities());
        this.principal = user;
        this.setAuthenticated(true);
    }

    public InternalAuthentication(UserAuthed user, Collection<? extends GrantedAuthority> authorities) {
        super(authorities);
        this.principal = user;
        this.setAuthenticated(true);
    }

    @Override
    public Object getCredentials() {
        return "";
    }

    @Override
    public Object getPrincipal() {
        return principal;
    }

}

File: services/archivist/archivist/src/main/kotlin/com/zorroa/archivist/security/InternalAuthentication.kt
package com.zorroa.archivist.security

import com.zorroa.archivist.clients.ApiKey
import org.springframework.security.authentication.AbstractAuthenticationToken
import org.springframework.security.core.GrantedAuthority

class InternalAuthentication : AbstractAuthenticationToken {

    private val principal: Any

    constructor(apiKey: ApiKey) : super(apiKey.getAuthorities()) {
        this.principal = apiKey
        this.isAuthenticated = true
    }

    constructor(apiKey: ApiKey, authorities: Collection<GrantedAuthority>) : super(authorities) {
        this.principal = apiKey
        this.isAuthenticated = true
    }

    override fun getCredentials(): Any {
        return ""
    }

    override fun getPrincipal(): Any {
        return principal
    }
}
